src/equations/ideal_glm_mhd_2d.jl

```diff
@@ -1119,6 +1119,30 @@ end
     return rho
 end
 
+@inline function velocity(u, equations::IdealGlmMhdEquations2D)
+    rho = u[1]
+    v1 = u[2] / rho
+    v2 = u[3] / rho
+    v3 = u[4] / rho
+    return SVector(v1, v2, v3)
+end
+
+@inline function velocity(u, orientation::Int, equations::IdealGlmMhdEquations2D)
+    rho = u[1]
+    v = u[orientation + 1] / rho
+    return v
+end
+
+@inline function velocity(u, normal_direction::AbstractVector,
+                          equations::IdealGlmMhdEquations2D)
+    rho = u[1]
+    v1 = u[2] / rho
+    v2 = u[3] / rho
+    v3 = u[4] / rho
+    v = v1 * normal_direction[1] + v2 * normal_direction[2] + v3 * normal_direction[3]
+    return v
+end
```

Comment:
This is a bit weird since we use all three velocity variables but only a 2D `normal_direction`. Thus, we can only use
```suggestion
@inline function velocity(u, normal_direction::AbstractVector,
                          equations::IdealGlmMhdEquations2D)
    rho = u[1]
    v1 = u[2] / rho
    v2 = u[3] / rho
    v = v1 * normal_direction[1] + v2 * normal_direction[2]
    return v
end
```

here. Correct me if I'm wrong, @andrewwinters5000

Reply:
Yes, that is the case.